### PR TITLE
[ISSUE #4326] Optimized code

### LINKED ISSUE #4326: Optimization code
Optimization code

**BUG REPORT**

1. Please describe the issue you observed:

- What did you do (The steps to reproduce)?

- What is expected to see?

- What did you see instead?

2. Please tell us about your environment:

3. Other information (e.g. detailed explanation, logs, related issues, suggestions on how to fix, etc):

### PR DESCRIPTION
**Pull request by Oliverwqcwrw**
**Make sure set the target branch to `develop`**

## What is the purpose of the change

Fix #4326 

Follow this checklist to help us incorporate your contribution quickly and easily. Notice, `it would be helpful if you could finish the following 5 checklist(the last one is not necessary)before request the community to review your PR`.

- [x] Make sure there is a [Github issue](https://github.com/apache/rocketmq/issues) filed for the change (usually before you start working on it). Trivial changes like typos do not require a Github issue. Your pull request should address just this issue, without pulling in other changes - one PR resolves one issue. 
- [x] Format the pull request title like `[ISSUE #123] Fix UnknownException when host config not exist`. Each commit in the pull request should have a meaningful subject line and body.
- [x] Write a pull request description that is detailed enough to understand what the pull request does, how, and why.
- [x] Write necessary unit-test(over 80% coverage) to verify your logic correction, more mock a little better when cross module dependency exist. If the new feature or significant change is committed, please remember to add integration-test in [test module](https://github.com/apache/rocketmq/tree/master/test).
- [x] Run `mvn -B clean apache-rat:check findbugs:findbugs checkstyle:checkstyle` to make sure basic checks pass. Run `mvn clean install -DskipITs` to make sure unit-test pass. Run `mvn clean test-compile failsafe:integration-test`  to make sure integration-test pass.
- [ ] If this contribution is large, please file an [Apache Individual Contributor License Agreement](http://www.apache.org/licenses/#clas).

### CODE REVIEW

**Review by li-xiao-shuang — APPROVED** [review]

**Comment by li-xiao-shuang** [comment]
LGTM

File: client/src/main/java/org/apache/rocketmq/client/consumer/DefaultMQPushConsumer.java [hunk 1]
```diff
@@ -417,7 +417,7 @@ public void createTopic(String key, String newTopic, int queueNum) throws MQClie
     @Override
     public void setUseTLS(boolean useTLS) {
         super.setUseTLS(useTLS);
-        if (traceDispatcher != null && traceDispatcher instanceof AsyncTraceDispatcher) {
```
Comment (hzh0425) [inline client/src/main/java/org/apache/rocketmq/client/consumer/DefaultMQPushConsumer.java:420]: I thanks 'traceDispatcher != null' is still needed

File: client/src/main/java/org/apache/rocketmq/client/producer/DefaultMQProducer.java [hunk 2]
```diff
@@ -896,7 +894,6 @@ public QueryResult queryMessage(String topic, String key, int maxNum, long begin
     public MessageExt viewMessage(String topic,
         String msgId) throws RemotingException, MQBrokerException, InterruptedException, MQClientException {
         try {
-            MessageId oldMsgId = MessageDecoder.decodeMessageId(msgId);
             return this.viewMessage(msgId);
         } catch (Exception e) {
         }
```
Comment (hzh0425) [inline client/src/main/java/org/apache/rocketmq/client/producer/DefaultMQProducer.java:899]: You can rename the 'Exception e' to 'Exception ignored'

File: client/src/main/java/org/apache/rocketmq/client/consumer/DefaultMQPushConsumer.java (hunk 1 above)
Comment (Oliverwqcwrw) [reply client/src/main/java/org/apache/rocketmq/client/consumer/DefaultMQPushConsumer.java:420]: IMO,Null is not an instanceof anything, so a null check is redundant.

File: client/src/main/java/org/apache/rocketmq/client/producer/DefaultMQProducer.java (hunk 2 above)
Comment (Oliverwqcwrw) [reply client/src/main/java/org/apache/rocketmq/client/producer/DefaultMQProducer.java:899]: Good idea

File: client/src/main/java/org/apache/rocketmq/client/consumer/DefaultMQPushConsumer.java (hunk 1 above)
Comment (li-xiao-shuang) [reply client/src/main/java/org/apache/rocketmq/client/consumer/DefaultMQPushConsumer.java:420]: > IMO,Null is not an instanceof anything, so a null check is redundant.

Agree with

File: client/src/main/java/org/apache/rocketmq/client/producer/DefaultMQProducer.java (hunk 2 above)
Comment (Oliverwqcwrw) [reply client/src/main/java/org/apache/rocketmq/client/producer/DefaultMQProducer.java:899]: Would it be better to print the exception message?

Comment (hzh0425) [reply client/src/main/java/org/apache/rocketmq/client/producer/DefaultMQProducer.java:899]: Oh, this method is 'Deprecated', just ignore it

Comment (Oliverwqcwrw) [reply client/src/main/java/org/apache/rocketmq/client/producer/DefaultMQProducer.java:899]: Ok,I will solve it

**Review by yuz10 — APPROVED** [review]

**Review by ShannonDing — APPROVED** [review]
LGTM

**Pull request merged**